Rideau/SwiftUI/Rideau+SwiftUI.swift

```diff
@@ -0,0 +1,441 @@
+
+    let controller = SwiftUISupports.RideauHostingController(
+      bodyViewController: context.coordinator.hostingController,
+      configuration: configuration,
+      resizingOption: .noResize,
+      usesDismissalPanGestureOnBackdropView: false,
+      hidesByBackgroundTouch: true,
+      onViewDidAppear: { viewController in
+        viewController.rideauView.move(to: initialSnapPoint, animated: true, completion: {})
+      }
+    )
+
+    controller.onDidDismiss = onDidDismiss
+
+    return controller
+
+  }
+
+  func updateUIViewController(
+    _ uiViewController: SwiftUISupports.RideauHostingController,
+    context: Context
+  ) {
+
+    context.coordinator.hostingController.rootView = content
+    
+  }
+
+}
+
+@available(iOS 14, *)
+private struct SwiftUIRideauItemModifier<Item: Identifiable, Body: View>: ViewModifier {
+
+  @Binding private var item: Item?
+  private let body: (Item) -> Body
+  private let onDismiss: () -> Void
+  private let configuration: RideauView.Configuration
+  private let initialSnapPoint: RideauSnapPoint
+
+  init(
+    configuration: RideauView.Configuration,
+    initialSnapPoint: RideauSnapPoint,
+    item: Binding<Item?>,
+    onDismiss: @escaping () -> Void,
+    body: @escaping (Item) -> Body
+  ) {
+    self.body = body
+    self.onDismiss = onDismiss
+    self._item = item
+    self.configuration = configuration
+    self.initialSnapPoint = initialSnapPoint
+
+  }
+
+  func body(content: Content) -> some View {
+    ZStack {
+      content
+      if let item {
+        SwiftUIRideau(
+          configuration: configuration,
+          initialSnapPoint: initialSnapPoint,
+          onDidDismiss: {
+            self.item = nil
+          },
+          conetnt: {
+            body(item)
+          }
+        )
+        .id(item.id)
+        .ignoresSafeArea()
+      }
+    }
+  }
+
+}
+
+@available(iOS 14, *)
+private struct SwiftUIRideauBooleanModifier<Body: View>: ViewModifier {
+
+  @Binding private var isPresented: Bool
+  private let body: Body
+  private let onDismiss: () -> Void
+  private let configuration: RideauView.Configuration
+  private let initialSnapPoint: RideauSnapPoint
+
+  init(
+    configuration: RideauView.Configuration,
+    initialSnapPoint: RideauSnapPoint,
+    isPresented: Binding<Bool>,
+    onDismiss: @escaping () -> Void,
+    body: Body
+  ) {
+    self.body = body
+    self._isPresented = isPresented
+    self.onDismiss = onDismiss
+    self.configuration = configuration
+    self.initialSnapPoint = initialSnapPoint
+
+  }
+
+  func body(content: Content) -> some View {
+    ZStack {
+      content
+      if isPresented {
+        SwiftUIRideau(
+          configuration: configuration,
+          initialSnapPoint: initialSnapPoint,
+          onDidDismiss: {
+            onDismiss()
+            isPresented = false
+          },
+          conetnt: {
+            body
+          }
+        )
+        .ignoresSafeArea()
+      }
+    }
+  }
+}
+
+#if DEBUG
```

Comment:
FYI in release build, PreviewProviders will not be included

Reply:
yeah, I know that. I just added macro for just in case of like adding a dummy struct inside macro.
even though, any types that are defined inside PreviewProvider, will be deleted in production. but not sure.
so still, adding macro makes me relieved.